epub33/core/index.html

```diff
@@ -1560,6 +1557,18 @@ <h6>The <code>metadata</code> Element</h6>
 
 						</section>
 
+						<section id="sec-package-elem-value">
+							<h5>Metadata Values</h5>
+
+							<p>When a child element of the <a href="#sec-metadata-elem"><code>metadata</code>
+									element</a> has mandatory text content, that content is referred to as the value of
+								the element in the explanatory descriptions.</p>
+
+							<p>These values MUST be at least one character in length after <a
+									href="https://www.w3.org/TR/epub-rs-33/#conf-meta-ws">stripping leading and trailing
+									ASCII whitespace</a> [[!EPUB-RS-33]].</p>
```

Comment:
I would put the "stripping leading and trailing" in the definition of the value, and put "value MUST be non-empty" statements case-by-case in the metadata property definitions.

If all metadata expressions mandate non-empty values, then the above definition of value can be simplified (no "When"), and the "value MUST be non-empty statement" can be defined for all.

Also, shouldn't "stripping leading and trailing" refer to INFRA rather than EPUB-RS here?

Reply:
> If all metadata expressions mandate non-empty values, then the above definition of value can be simplified (no "When"), and the "value MUST be non-empty statement" can be defined for all.

Right, this is what being more specific might help with. There is a finite set of elements and they all require a non-empty value. That's why I'd rather define this once than list it for each. It's always been cumbersome.

> Also, shouldn't "stripping leading and trailing" refer to INFRA rather than EPUB-RS here?

I considered that, but it's technically the RS that does the stripping. If you want to find out more, the first hop should be to the RS spec.

Reply:
> > Also, shouldn't "stripping leading and trailing" refer to INFRA rather than EPUB-RS here?
> 
> I considered that, but it's technically the RS that does the stripping. I

Even if the RS will effectively do the stripping at some point, the stripping is a fundamental part of the MUST statement (or of the definition of "value", alternatively).

As an example (out of thin air 😇😁),  a checker would need to do the stripping too, it shouldn't have to go through the RS spec indirection to interpret this MUST statement 😊.

Reply:
> a checker would need to do the stripping too, it shouldn't have to go through the RS spec indirection to interpret this MUST statement

I'd argue it does, though, as the stripping is conditional.

For DC elements its required, but there's a loophole for the meta element that properties can override the default behaviour. This is underspecified, in my opinion, as I believe the intention was only that properties _we_ define can override, not that any property defined in any vocabulary can change our rules. That would be a nightmare of inconsistency to deal with.

But given that I'm not aware of any properties we define that change the whitespace handling rules, I wonder if we can just do away with that rule entirely?

I'd then be more at ease with each spec separately referencing across to infra.